### ISSUE TITLE
Unable to Delete Files to Trash: vscode( V > 1.89 )  (on debian kde)

### ISSUE THREAD

**Issue by sonk33** [microsoft/vscode#214728]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.90.0-1717531825_amd64
- OS Version: debian 12
- Distro: Debian GNU/Linux trixie/sid
- Kernel: 6.8.12-amd64 arch: x86_64 bits: 64 compiler: gcc v: 13.2.0
- Desktop: KDE Plasma v: 5.27.11 - wayland 

**Steps to Reproduce**:
1. Disable all extensions (Not exactly relevant)
2. Attempt to delete any file to the trash.

**Expected Behavior**:
  When attempting to delete a file, VSCode should delete it to the trash without any issues.

**Actual Behavior**:
  VSCode keeps trying to delete the file endlessly without success.

**Additional Information**:

I tracked what VSCode was trying to do. It launches this command: `kioclient5 move /path/file_name.any trash:/`
and this command keeps running endlessly without success.

So, I used the same command in the system terminal (Konsole), and it works fine, and the file was deleted successfully. 
Then, I used the same command in VSCode's integrated terminal, but it runs endlessly.

I used strace to trace the call, and here are the results:

**the result of runing** `strace -f -e trace=execve kioclient5 move /home/userName/path_longPath/file.any trash:/`
<details>
<summary>kde konsole </summary>
execve("/usr/bin/kioclient5", ["kioclient5", "move", "/home/userName/path_lo"..., "trash:/"], 0x7fff0da66a70 /* 66 vars */) = 0

strace: Process 102239 attached
strace: Process 102240 attached
strace: Process 102241 attached
strace: Process 102242 attached
strace: Process 102243 attached
[pid 102243] +++ exited with 0 +++
[pid 102242] +++ exited with 0 +++
strace: Process 102244 attached
strace: Process 102245 attached
strace: Process 102246 attached
[pid 102246] execve("/lib/x86_64-linux-gnu/libexec/kf5/kioslave5", ["/lib/x86_64-linux-gnu/libexec/kf"..., "/usr/lib/x86_64-linux-gnu/qt5/pl"..., "trash", "", "local:/run/user/1000/kioclientvu"...], 0x7ffde30daaa0 /* 66 vars */ <unfinished ...>
[pid 102245] +++ exited with 1 +++
[pid 102238] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=102245, si_uid=1000, si_status=1, si_utime=0, si_stime=0} ---
[pid 102246] <... execve resumed>)      = 0
strace: Process 102247 attached
strace: Process 102248 attached
[pid 102246] --- SIGTERM {si_signo=SIGTERM, si_code=SI_USER, si_pid=102238, si_uid=1000} ---
[pid 102247] --- SIGRTMIN {si_signo=SIGRTMIN, si_code=SI_TKILL, si_pid=102246, si_uid=1000} ---
[pid 102247] +++ exited with 0 +++
[pid 102248] +++ exited with 0 +++
[pid 102246] +++ exited with 0 +++
[pid 102241] +++ exited with 0 +++
[pid 102239] +++ exited with 0 +++
[pid 102240] +++ exited with 0 +++
[pid 102244] +++ exited with 0 +++
+++ exited with 0 +++
</details>
<details>
<summary>vscode default integrated terminal </summary>

execve("/usr/bin/kioclient5", ["kioclient5", "move", "/home/userName/path_lo"..., "trash:/"], 0x7fff9778abe0 /* 66 vars */) = 0

strace: Process 102649 attached
strace: Process 102650 attached
strace: Process 102651 attached
strace: Process 102652 attached
strace: Process 102653 attached
[pid 102653] +++ exited with 0 +++
[pid 102652] +++ exited with 0 +++
strace: Process 102654 attached
strace: Process 102655 attached
strace: Process 102656 attached

[pid 102656] execve("/lib/x86_64-linux-gnu/libexec/kf5/kioslave5", ["/lib/x86_64-linux-gnu/libexec/kf"..., "/usr/lib/x86_64-linux-gnu/qt5/pl"..., "trash", "", "local:/run/user/1000/kioclientxf"...], 0x7ffc2cd588f0 /* 66 vars */ <unfinished ...>
[pid 102655] +++ exited with 1 +++
[pid 102648] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=102655, si_uid=1000, si_status=1, si_utime=0, si_stime=0} ---
[pid 102656] <... execve resumed>)      = 0

i had to kill it (ctr+c) to make it end.
</details>

In short, when trying to execute the command in VSCode, it hangs at this step: 
`/lib/x86_64-linux-gnu/libexec/kf5/kioslave5 /usr/lib/x86_64-linux-gnu/qt5/plugins/kf5/kio/kio_trash.so trash local:/run/user/1000/kioclientfXUcta.1.kiowork`

After reverting to the previous version of VSCode (1.89.0-1714530869_amd64), the issue was resolved.

**Comment by arsinclair** [comment]
For me the deletion hangs for a number 20-30 seconds but the file is deleted in the end. The `kioslave5` CPU usage rises to 80-100% during this operation.

**Comment by nohappiness** [comment]
Confirm the issue here, system info and vscode info as below.

BTW, for those bothered by this issue, you can temporarily disable trash to bypass the long hangs on "moving file to trash..." until this got fixed here or by upstream(electron).

> Operating System: Debian GNU/Linux 12
> KDE Plasma Version: 5.27.11
> KDE Frameworks Version: 5.115.0
> Qt Version: 5.15.13
> Kernel Version: 6.11.4-amd64 (64-bit)
> Graphics Platform: X11
> Processors: 16 × 13th Gen Intel® Core™ i5-13500H
> Memory: 62.5 GiB of RAM
> Graphics Processor: Mesa Intel® Graphics
> Manufacturer: HP
> Product Name: Victus by HP Gaming Laptop 15-fa1xxx

VSCode: 

> 版本: 1.94.2
> 提交: 384ff7382de624fb94dbaf6da11977bba1ecd427
> 日期: 2024-10-09T16:08:44.566Z
> Electron: 30.5.1
> ElectronBuildId: 10262041
> Chromium: 124.0.6367.243
> Node.js: 20.16.0
> V8: 12.4.254.20-electron.0
> OS: Linux x64 6.11.4-amd64

**Comment by sh0z3n** [comment]
set this into you settings on vscode , it should disable that feature 

```
{
    "files.enableTrash": false
}

```

**still the issue is more related to the API of   _Electron js_  the responsible to take files into the path of trash ( it can be diffrent from an os to the other and especially with KDE **

**Comment by arsinclair** [comment]
That is a workaround and not a solution, I guess we'd like the files to go to the Trash instead of being permanently deleted if we're experiencing this issue.

**Comment by sh0z3n** [comment]
> That is a workaround and not a solution, I guess we'd like the files to go to the Trash instead of being permanently deleted if we're experiencing this issue.

i literally said : **still the issue is more related to the API of Electron js the responsible to take files into the path of trash ( it can be diffrent from an os to the other and especially with KDE**

**Comment by arsinclair** [comment]
Yes I saw that part. I'm only referring to your code snippet, because it doesn't mention that this is a workaround for the issue, and also this workaround was already mentioned by the previous commented.

> BTW, for those bothered by this issue, you can temporarily disable trash(in settings) to bypass the long hangs on "moving file to trash..." until this got fixed here or by upstream(electron)
